src/rov/rov_pca9685/src/PCA9685.cpp

```diff
@@ -0,0 +1,190 @@
+#include "pca9685/PCA9685.hpp"
+
+extern "C" {
+#include "linux/i2c-dev.h"
+#include "i2c/smbus.h"
+}
+
+#include <cstddef>
+#include <sys/ioctl.h>
+#include <cstdlib>
+#include <cstdio>
+#include <fcntl.h>
+#include <unistd.h>
+#include <errno.h>
+
+#include <cmath>
+
+#define US_PER_S 1e6
+
+PCA9685::PCA9685(uint8_t bus, uint8_t address) : i2c_address(address), i2c_bus(bus) {
+    clock_frequency = 25e6;
+    // 1e6 us per s / 25e6 hz = 0.04 us
+    clock_us = US_PER_S / 25e6;
+    output_frequency = 200;
+}
+
+PCA9685::~PCA9685() {
+    Close();
+}
+
+bool PCA9685::Open() {
+    char filename[32];
+    sprintf(filename,"/dev/i2c-%d", i2c_bus);
+    // printf("%s\n", filename);
+    i2c_fd = open(filename, O_RDWR);
+    if (i2c_fd < 0) {
+        // Could not open the bus
+    
+        return false;
+    }
+    if (ioctl(i2c_fd, I2C_SLAVE, i2c_address) < 0) {
+        // Could not open the device on the bus
+        return false;
+    }
+
+    // external clock?
+    // i2c_smbus_write_byte_data(i2c_fd, MODE1, MODE1_SLEEP);
```

Comment:
Ew! Yucky comments that weren't deleted before committing!

Reply:
left in as this is how to enable the external clock and we will be getting one of those on the pip 2.0